### PR TITLE
added monochrome, 8bpp indexed, and 32bpp ARGB BMPs

### PR DESCRIPTION
**Pull request by siddacious**
More bitdepths! Tested on PyPortal.

### CODE REVIEW

**Comment by ladyada** [comment]
ive tested this, works gr8

File: shared-module/displayio/OnDiskBitmap.c [hunk 1]
```diff
@@ -100,39 +123,54 @@ uint32_t common_hal_displayio_ondiskbitmap_get_pixel(displayio_ondiskbitmap_t *s
...
+        location = self->data_offset + (self->height - y) * self->stride + x * bytes_per_pixel;
+    } else {
+        location = self->data_offset + (self->height - y) * self->stride + x / 8;
+    }
     // We don't cache here because the underlying FS caches sectors.
     f_lseek(&self->file->fp, location);
     UINT bytes_read;
-    uint32_t pixel = 0;
-    uint32_t result = f_read(&self->file->fp, &pixel, self->bytes_per_pixel, &bytes_read);
+    uint32_t pixel = 0; // this name is stale
```
Comment (dhalbert) [inline shared-module/displayio/OnDiskBitmap.c]: Not sure what you mean by stale.

File: shared-module/displayio/OnDiskBitmap.c [hunk 2]
```diff
@@ -100,39 +123,54 @@ uint32_t common_hal_displayio_ondiskbitmap_get_pixel(displayio_ondiskbitmap_t *s
...
+        } else if (bytes_per_pixel == 1){ 
+            blue = ((self->palette_data[pixel] & 0xFF) >> 0);
+            red = ((self->palette_data[pixel] & 0xFF0000) >> 16);
+            green = ((self->palette_data[pixel] & 0xFF00) >> 8);
+            tmp = (red << 16 | green << 8 | blue );
             return tmp;
-        } else if (self->bytes_per_pixel == 2) {
-            if (self->bitfield_compressed){
+        } else if (bytes_per_pixel == 2) {
+            if (self->g_bitmask == 0x07e0){ // 565
```
Comment (dhalbert) [inline shared-module/displayio/OnDiskBitmap.c]: Add space before `{`.

File: shared-module/displayio/OnDiskBitmap.c [hunk 3]
```diff
@@ -100,39 +123,54 @@ uint32_t common_hal_displayio_ondiskbitmap_get_pixel(displayio_ondiskbitmap_t *s
...
-            } else {
+            } else { // 555
                 red =((pixel & self->r_bitmask) >>10);
                 green = ((pixel & self->g_bitmask) >>4);
                 blue = ((pixel & self->b_bitmask) >> 0);
             }
             tmp = (red << 19 | green << 10 | blue << 3);
             return tmp;
-        }else {
+        }else if ((bytes_per_pixel == 4) && (self->bitfield_compressed)){
```
Comment (dhalbert) [inline shared-module/displayio/OnDiskBitmap.c]: Add space after `}` and before `{`.

File: shared-module/displayio/OnDiskBitmap.c [hunk 4]
```diff
@@ -56,42 +56,65 @@ void common_hal_displayio_ondiskbitmap_construct(displayio_ondiskbitmap_t *self,
...
-    } else if (!(header_size == 12 || header_size == 40 || header_size == 108 || header_size == 124) ||
-        !(compression == 0)) {
+    } else if (!(header_size == 12 || header_size == 40 || header_size == 108 || header_size == 124)) {
         mp_raise_ValueError_varg(translate("Only Windows format, uncompressed BMP supported %d"), header_size);
     }
 
-    if (bits_per_pixel < 16 && !(self->grayscale)) {
-        mp_raise_ValueError_varg(translate("Only true color (24 bpp or higher), 16bpp 565 and 555, and 8bpp grayscale BMP supported %d"), bits_per_pixel);
+    if ((bits_per_pixel == 4 ) || (( bits_per_pixel == 8) && (number_of_colors == 0))) {
+        mp_raise_ValueError_varg(translate("Only monochrome, indexed 8bpp, and 16bpp or greater BMPs supported: %d"), bits_per_pixel);
```
Comment (dhalbert) [inline shared-module/displayio/OnDiskBitmap.c]: Can you change this message to be clearer about what the printed numeric value is, like "...supported, not %d bpp" or something like that.

File: shared-module/displayio/OnDiskBitmap.c [hunk 5]
```diff
@@ -133,43 +133,43 @@ uint32_t common_hal_displayio_ondiskbitmap_get_pixel(displayio_ondiskbitmap_t *s
     // We don't cache here because the underlying FS caches sectors.
     f_lseek(&self->file->fp, location);
     UINT bytes_read;
-    uint32_t pixel = 0; // this name is stale
-    uint32_t result = f_read(&self->file->fp, &pixel, bytes_per_pixel, &bytes_read);
+    uint32_t pixel_data = 0; // this name is stale
```
Comment (dhalbert) [inline shared-module/displayio/OnDiskBitmap.c]: Remove "this name is stale".

Comment (siddacious) [reply shared-module/displayio/OnDiskBitmap.c]: 🤦‍♂️

**Comment by dhalbert** [comment]
Thank you!

**Pull request merged**